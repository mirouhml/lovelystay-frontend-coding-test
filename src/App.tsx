import React from 'react';
import { Route, Routes } from 'react-router-dom';
import DisplayUserPage from './pages/DisplayUserPage';
import SearchUserPage from './pages/SearchUserPage';
import NotFoundPage from './pages/NotFoundPage';

function App() {
  return (
    <main>
      <Routes>
        <Route path='/' element={<SearchUserPage />} />
        <Route path='/:username' element={<DisplayUserPage />} />
        <Route path='*' element={<NotFoundPage />} />
      </Routes>
    </main>
  );
}

export default App;
