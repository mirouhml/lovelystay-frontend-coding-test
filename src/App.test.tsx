import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from './App';
import store from './app/store';

describe('App', () => {
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/test/not-a-real-page']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have a search page as the home page', () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(screen.getByText('Search for a GitHub user')).toBeInTheDocument();
  });

  it('should have a 404 page', () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/test/not-a-real-page']}>
          <App />
        </MemoryRouter>
      </Provider>
    );
    expect(screen.getByText('404 - Page Not Found')).toBeInTheDocument();
  });
});
