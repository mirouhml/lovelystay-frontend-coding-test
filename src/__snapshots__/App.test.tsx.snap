// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main>
  <div
    className="not-found-page"
  >
    <h1>
      404 - Page Not Found
    </h1>
  </div>
</main>
`;
